src/submission/submission-to-clinical/merge-submission.ts

```diff
@@ -111,15 +111,15 @@ export const mergeActiveSubmissionWithDonors = async (
   return updatedDonors;
 };
 
-// This function will return a merged donor of records mapped by clinical type and the DB exsistentDonor
+// This function will return a merged donor of records mapped by clinical type and the DB existentDonor
 export const mergeRecordsMapIntoDonor = (
   unmutableSubmittedRecordsMap: DeepReadonly<SubmittedClinicalRecordsMap>,
-  exsistentDonor: DeepReadonly<Donor>,
+  existentDonor: DeepReadonly<Donor>,
 ) => {
   const submittedRecordsMap = _.cloneDeep(
     unmutableSubmittedRecordsMap,
   ) as SubmittedClinicalRecordsMap;
-  const mergedDonor: any = _.cloneDeep(exsistentDonor);
+  const mergedDonor: any = _.cloneDeep(existentDonor) || {};
```

Comment:
is there a better fix for this than using `any`?
It looks like a workaround for `DeepReadOnly<obj>` vs `obj`

I know this issue is all over the codebase but I think it's a good opportunity to figure out a fix or at least a better bandage than `any` once and for all.

these `DeepReadOnly` types come from the `deep-freeze` package which is actually a straightforward singular function + typings. It just iterates over an object and recursively freezes it's properties using the js `freeze` function. Unfortunately in js this is the ultimate in locking things up ... there is no "unfreeze" (at least that I'm aware of).  Best way to work through this is to clone the object I think... which we are doing... but the frozen object types still follow. This is an issue because there is no enforcement of the freeze despite what the types are saying. You can see in this function the cloned object properties get mutated. 

I'll add a follow up comment as I'm currently trying to figure out the best approach here but wanted to get it started for any input you have

Reply:
I think this is actually one of the few times using an assertion might be a valid case?
```
  const mergedDonor = (_.cloneDeep(existentDonor) || {}) as Donor;
```

we trust that `cloneDeep` is giving us a deep clone and it's a new object, and it's mutable. so we can tell TS that via the assertion.

Reply:
Comment #1: So the main purpose of this section is to compare `const treatmentInvalid = totalTreatmentTime > submittedInterval;` where `totalTreatmentTime` and `submittedInterval` should be numbers

`ClinicalInfo` is typed extremely generic, so there is no Type guarantee for variables such as `interval_of_followup`, `treatment_start_interval` and `treatment_duration`.

I actually think this section should be written as:

` const prevIntervalOfFollowUp =
        (typeof prevEntityIdMatch?.clinicalInfo?.interval_of_followup === 'number' &&
          prevEntityIdMatch.clinicalInfo.interval_of_followup) ||
        0;

      const submittedIntervalOfFollowUp =
        (typeof submittedIdMatch?.interval_of_followup === 'number' &&
          submittedIdMatch?.interval_of_followup) ||
        0;

      const submittedInterval = prevIntervalOfFollowUp
        ? prevIntervalOfFollowUp
        : submittedIntervalOfFollowUp;

      const prevInvalidTreatments = prevTreatments.filter(treatment => {
        const { treatment_start_interval, treatment_duration } = treatment.clinicalInfo
          ? treatment.clinicalInfo
          : treatment;

        const treatmentInterval =
          typeof treatment_start_interval === 'number' ? treatment_start_interval : 0;

        const treatmentDuration = typeof treatment_duration === 'number' ? treatment_duration : 0;

        const totalTreatmentTime = treatmentInterval + treatmentDuration;

        const treatmentInvalid = totalTreatmentTime > submittedInterval;`
        
        This guarantees the last line is comparing numberA > numberB

Reply:
see: https://github.com/icgc-argo/argo-clinical/pull/1002/commits/1d9e471ca8f65e3a68f2dbdd0b03276b679106be